Document implementation of new service (#6)
vshn/application-catalog-docs: docs/modules/ROOT/pages/how-tos/implement_new_service_offering.adoc

* `service.syn.tools/id`: UUID of the service this plan belongs to
* `service.syn.tools/name`: Name of the service this plan belongs to
* `service.syn.tools/updatable`: If the service instances can be updated

To add further metadata you can use the following annotations:

* `service.syn.tools/description`: Description for this plan
* `service.syn.tools/metadata`: JSON object with further https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#service-plan-object[metadata]

==== Resources
A Composition defines a list of K8s resources which should be created for each instance of this plan.
These resources together build the provisioned instance and usually consist of `Release` resources which the https://github.com/crossplane-contrib/provider-helm[provider-helm] then installs accordingly.

Patches can be used to further parametrize the resources and use information on the XR as input for further templating.

==== Example
[source,yaml]
----
apiVersion: apiextensions.crossplane.io/v1
kind: Composition
metadata:
  name: 43905323-ed5f-466a-9d7f-ddc16cb74864 # uuidgen | tr '[:upper:]' '[:lower:]'
  labels:
    service.syn.tools/cluster: prod-cluster-01

>     service.syn.tools/id: 6ca63cdb-0cfa-4c5e-b080-72f22ff5f3e6

Reviewer: ```suggestion
    service.syn.tools/id: 6ca63cdb-0cfa-4c5e-b080-72f22ff5f3e6 # This is the ID from 'compositefooinstances.syn.tools' above
```

          toFieldPath: spec.forProvider.namespace
        - fromFieldPath: metadata.labels[service.syn.tools/cluster]
          toFieldPath: spec.providerConfigRef.name
        - fromFieldPath: metadata.labels[crossplane.io/composite]
          toFieldPath: spec.writeConnectionSecretToRef.name
          transforms:
            - string:
                fmt: "%s-foo"
              type: string
        - fromFieldPath: metadata.labels[crossplane.io/composite]
          toFieldPath: spec.connectionDetails[0].namespace
        - fromFieldPath: metadata.labels[crossplane.io/composite]
          toFieldPath: spec.connectionDetails[1].namespace
        - fromFieldPath: metadata.labels[crossplane.io/composite]
          toFieldPath: spec.connectionDetails[2].namespace
        - fromFieldPath: metadata.labels[crossplane.io/composite]
          toFieldPath: spec.connectionDetails[3].namespace
----

=== Commodore Component

To simplify the creation of these Crossplane resources the https://git.vshn.net/swisscompks/component-spks-crossplane[spks-crossplane component] exists.
It will generate the required resources based on the configured input in the configuration hierarchy.

> Especially to define multiple plans this approach helps to keep the config more DRY and maintainable.

Reviewer: ```suggestion
This approach especially helps in defining multiple plans in order to keep the config more DRY and maintainable.
```

==== Example

> This example setup will generate the same Crossplane resoruces as showcased in the previous examples.

Reviewer: ```suggestion
This example setup will generate the same Crossplane resources as showcased in the previous examples.
```
